### PR TITLE
Add Moat grapple teleport strats

### PR DESCRIPTION
**Pull request by blkerby**
In addition to the strats for Moat, this also adds a few more strats from Red Brinstar Firefleas that were skipped earlier:
- into Waterway (end up stuck behind Speed blocks)
- into X-Ray Scope (basically needs the item to be Morph)
- into Green Pirates Shaft (needs a jump like from Moat, and then a Crystal Flash)

It also includes a couple fixes:
- in Acid Statue Room, the statue cannot be used while off-camera; so an obstacle is now used to track being off-camera.
- the teleport into Purple Shaft from Red Brinstar Firefleas requires some unintuitive off-camera movement (extending and then retracting Grapple), otherwise a much larger amount of heat damage would be required. So this is now reflected in the requirement of either `canOffScreenMovement` or extra `heatFrames`.

### CODE REVIEW

**Review by osse101 — COMMENTED** [review]
It takes me about 15 tries to get each of these.  There's barely not enough time for me to react to getting the landing shot so I know to moonwalk or not.

File: region/brinstar/red/Below Spazer.json [hunk 1]
```diff
@@ -301,6 +301,18 @@
...
+    {
+      "link": [2, 3],
+      "name": "Grapple Teleport",
+      "entranceCondition": {
+        "comeInWithGrappleTeleport": {
+          "blockPositions": [[7, 2]]
+        }
+      },
+      "requires": [
+        "h_canBombThings"
```
[the remark this answers is not in the document]
Comment (blkerby) [reply region/brinstar/red/Below Spazer.json:313]: Good catch, I missed the lower-right entrance on this. Added now.

File: region/lowernorfair/west/Acid Statue Room.json [hunk 2]
```diff
@@ -306,10 +314,36 @@
...
       "note": [
         "Grapple teleporting here will spawn Samus inside the wall behind the Acid Chozo statue.",
         "To escape, perform a Crystal Flash to stand up, then morph and roll out to the right.",
-        "Samus will be visible but off-camera, making the movement tricky."
+        "Samus will be visible but off-camera, making the movement tricky.",
+        "While off camera, the Chozo hand will not function (i.e., will not trigger the acid lowering cutscene)."
+      ]
+    },
+    {
+      "link": [1, 3],
```
[the remark this answers is not in the document]
Comment (blkerby) [reply region/lowernorfair/west/Acid Statue Room.json:326]: Good catch, that was the one I thought I was doing; forgot about the other entrance. Added now.

File: region/maridia/inner-pink/Aqueduct.json [hunk 3]
```diff
@@ -1392,7 +1392,7 @@
       "name": "Grapple Teleport",
       "entranceCondition": {
         "comeInWithGrappleTeleport": {
-          "blockPositions": [[5, 3]]
+          "blockPositions": [[5, 3], [7, 2]]
```
[the remark this answers is not in the document]
Comment (blkerby) [reply region/maridia/inner-pink/Aqueduct.json]: Ah, missed that this was possible. If you press down after teleporting but before the door finishes opening, then you can swing up above the bomb blocks, which I guess must be what I was doing before. So you can get to either side of them. Added this other option and a note about how to get above the bomb blocks.

[the remark this answers is not in the document]
Comment (osse101) [reply region/maridia/inner-pink/Aqueduct.json]: I see that I'm getting blue doored.  Shame you can't buffer the down input.

**Pull request merged**